rhoc/pkg/cmd/deployments/snapshot/snapshot.go

```diff
@@ -0,0 +1,176 @@
+package snapshot
+
+import (
+	"errors"
+	"fmt"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/service"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/cmdutil"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/dumper"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/request"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/spf13/cobra"
+	"time"
+)
+
+const (
+	CommandName  = "snapshot"
+	CommandAlias = "ss"
+)
+
+type options struct {
+	clusterAmount int
+	outputFile    string
+
+	f *factory.Factory
+}
+
+func NewSnapshotCommand(f *factory.Factory) *cobra.Command {
+	opts := options{
+		f: f,
+	}
+
+	cmd := &cobra.Command{
+		Short:   "Takes a snapshot of deployments in oldest and ready clusters.",
+		Use:     CommandName,
+		Aliases: []string{CommandAlias},
+		Args:    cobra.NoArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if opts.clusterAmount < 1 || opts.clusterAmount > 30 {
+				return errors.New("clusterAmount must be between 1 and 30")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(&opts)
+		},
+	}
+
+	cmdutil.AddClusterAmount(cmd, &opts.clusterAmount)
```

Comment:
maybe there should be a way to define for which cluster you want to get the  snapshot ? my understading is that you can define a number of cluster for which you want a snapshot but that would be unreliable

Reply:
the reasoning was querying for X clusters with status=ready and sorted by age.
Right now I'd rather change the validation mechanism to parse the CSV and query for the same clusters afterwards, as to prevent that from happening.

adding the option to set specific clusters is a valid option and can also be done, I just wanted something more straightforward than figuring out know which are the "important clusters" beforehand.